libexec/scoop-search.ps1

```diff
@@ -4,119 +4,187 @@
 #
 # If used with [query], shows app names that match the query.
 # Without [query], shows all the available apps.
-param($query)
-. "$psscriptroot\..\lib\core.ps1"
-. "$psscriptroot\..\lib\buckets.ps1"
+param(
+    [Parameter(Mandatory = $true)]
+    $Query,
+    [Switch] $Remote
+)
 . "$psscriptroot\..\lib\manifest.ps1"
+. "$psscriptroot\..\lib\install.ps1"
 . "$psscriptroot\..\lib\versions.ps1"
 . "$psscriptroot\..\lib\config.ps1"
 
 reset_aliases
 
-function bin_match($manifest, $query) {
-    if(!$manifest.bin) { return $false }
-    foreach($bin in $manifest.bin) {
-        $exe, $alias, $args = $bin
-        $fname = split-path $exe -leaf -ea stop
+function search_bucket($bucket, $query) {
+    $arch = default_architecture
+    $apps = apps_in_bucket (Find-BucketDirectory $bucket) | ForEach-Object {
+        $manifest = manifest $_ $bucket
+        @{
+            name = $_
+            version = $manifest.version
+            description = $manifest.description
+            shortcuts = @(arch_specific 'shortcuts' $manifest $arch)
+            matchingShortcuts = @()
+            bin = @(arch_specific 'bin' $manifest $arch)
+            matchingBinaries = @()
+        }
+    }
 
-        if((strip_ext $fname) -match $query) { return $fname }
-        if($alias -match $query) { return $alias }
+    if(!$query) {
+        return $apps
     }
-    $false
-}
 
-function search_bucket($bucket, $query) {
-    $apps = apps_in_bucket (Find-BucketDirectory $bucket) | ForEach-Object {
-        @{ name = $_ }
+    try {
+        $query = New-Object Regex $query, 'IgnoreCase'
+    } catch {
+        abort "Invalid regular expression: $($_.exception.innerexception.message)"
     }
```

Comment:
I would move this within function. When it will be standalone inside `scoop-search.ps1` you will save abort call and can handle failsafe scenario better.

Then the query, which is passing into `search_bucket` function could be Regex object

Reply:
Also you do not need to create regex for each bucket. This regex is same for all buckets